enhancements/20220720-alertmanager.md

```diff
@@ -0,0 +1,819 @@
+      "generatorURL": <string>,      // identifies the entity that caused the alert
+      "fingerprint": <string>        // fingerprint to identify the alert
+    },
+    ...
+  ]
+}
+```
+
+##### add/delete `receiver` via CRD `AlertmanagerConfig`
+
+By CRD `AlertmanagerConfig`, user can config the receiver.
+
+The default `Alertmanager` config, it has a global `null` receiver, no alert will be sent out of of `Alertmanager`.
+
+```
+/alertmanager $ cat /etc/alertmanager/config/alertmanager.yaml 
+
+global:
+  resolve_timeout: 5m
+route:
+  receiver: "null"
+  group_by:
+  - job
+  routes:
+  - receiver: "null"
+    match:
+      alertname: Watchdog
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 12h
+receivers:
+- name: "null"
+templates:
+- /etc/alertmanager/config/*.tmpl
+```
+
+An example, the input is
+
+```
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: AlertmanagerConfig
+metadata:
+  name: amc-example
+  labels:
+    alertmanagerConfig: example
+spec:
+  route:
+    continue: true
+    groupWait: 30s
+    groupInterval: 5m
+    repeatInterval: 12h
+    groupBy:
+    - cluster
+    - alertname
+    receiver: "test-py-receiver"
+    routes:
+    - receiver: "amc-webhook-reciver"
+      groupBy: ['job']
+      groupWait: 30s
+      groupInterval: 5m
+      repeatInterval: 12h
+  receivers:
+  - name: "test-py-receiver"
+    webhookConfigs:
+    - sendResolved: true
+      url: "http://192.168.122.159:8080/"
+  - name: blackhole
+  - name: "amc-webhook-reciver"
+    webhookConfigs:
+    - sendResolved: true
+      url: "http://192.168.122.159:8090/"
+```
+
+Apply such an YAML, an CRD object is created.
+
+```
+# kubectl get AlertmanagerConfig -A
+NAMESPACE   NAME          AGE
+default     amc-example   10m
+```
+
+
+The updated `Alertmanager` config:
+
+```
+/alertmanager $ cat /etc/alertmanager/config/alertmanager.yaml 
+
+global:
+  resolve_timeout: 5m
+route:
+  receiver: "null"
+  group_by:
+  - job
+  routes:
+  - receiver: default-amc-example-test-py-receiver
+    group_by:
+    - cluster
+    - alertname
+    match:
+      namespace: default
+    continue: true
+    routes:
+    - receiver: default-amc-example-amc-webhook-reciver
+      group_by:
+      - job
+      group_wait: 30s
+      group_interval: 5m
+      repeat_interval: 12h
+    group_wait: 30s
+    group_interval: 5m
+    repeat_interval: 12h
+  - receiver: "null"
+    match:
+      alertname: Watchdog
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 12h
+receivers:
+- name: "null"
+- name: default-amc-example-test-py-receiver
+  webhook_configs:
+  - send_resolved: true
+    url: http://192.168.122.159:8080/
+- name: default-amc-example-blackhole
+- name: default-amc-example-amc-webhook-reciver
+  webhook_configs:
+  - send_resolved: true
+    url: http://192.168.122.159:8090/
+templates:
+- /etc/alertmanager/config/*.tmpl
+```
+
+[TBD, more tests are onging, will have a final summary]
+
+##### add/delete `receiver` via managedchart
```

Comment:
I've tested `AlertmanagerConfig` and it works.
Should we remove the ManagedChart hack here if AlertMangerConfig works?

Reply:
Updated and enhanced the `AlertmanagerConfig` chapter.

Remark this part as `debug only`, we can not stop others from editing `managedchart`, would like to keep it for better understanding and easy trouble-shooting.